src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-features-count/ProteinFeaturesCount.tsx

```diff
@@ -14,82 +14,71 @@
-    fetchProteinSummaryStats(props.transcriptId).then((response) => {
-      setProteinStats(response);
-    });
-  }, [props.transcriptId]);
+enum FeatureCountLabel {
+  ANNOTATIONS = 'Functional annotations',
+  INTERACTIONS = 'Interactions',
+  LIGANDS = 'Ligands',
+  STRUCTURES = 'Structures'
+}
 
-  if (!proteinStats) {
-    return null;
-  }
-
-  const midStyles = classNames(transcriptListStyles.middle, styles.middle);
+const ProteinFeaturesCount = (props: ProteinFeaturesCountProps) => {
+  const { proteinStats } = props;
 
   return (
-    <div className={transcriptListStyles.row}>
-      <div className={transcriptListStyles.left}>PDBe-KB P51587</div>
-      <div className={midStyles}>
-        <div className={styles.feature}>
-          <div className={styles.featureImg}>
-            <img src={structuresIcon} alt="" />
-          </div>
-          <div className={styles.featureCount}>
-            {proteinStats.structuresCount}
-          </div>
-          <div className={styles.featureText}>Structures</div>
-        </div>
-        <div className={styles.feature}>
-          <div className={styles.featureImg}>
-            <img src={ligandsIcon} alt="" />
-          </div>
-          <div className={styles.featureCount}>{proteinStats.ligandsCount}</div>
-          <div className={styles.featureText}>Ligands</div>
-        </div>
-        <div className={styles.feature}>
-          <div className={styles.featureImg}>
-            <img src={interactionsIcon} alt="" />
-          </div>
-          <div className={styles.featureCount}>
-            {proteinStats.interactionsCount}
-          </div>
-          <div className={styles.featureText}>Interactions</div>
-        </div>
-        <div className={styles.feature}>
-          <div className={styles.featureImg}>
-            <img src={annotationsIcon} alt="" />
-          </div>
-          <div className={styles.featureCount}>
-            {proteinStats.annotationsCount}
-          </div>
-          <div className={styles.featureText}>Functional annotations</div>
-        </div>
-      </div>
+    <div>
+      <FeatureCount
+        label={FeatureCountLabel.STRUCTURES}
+        count={proteinStats.structuresCount}
+        icon={structuresIcon}
+      />
+      <FeatureCount
+        label={FeatureCountLabel.LIGANDS}
+        count={proteinStats.ligandsCount}
+        icon={ligandsIcon}
+      />
+      <FeatureCount
+        label={FeatureCountLabel.INTERACTIONS}
+        count={proteinStats.interactionsCount}
+        icon={interactionsIcon}
+      />
+      <FeatureCount
+        label={FeatureCountLabel.ANNOTATIONS}
+        count={proteinStats.annotationsCount}
+        icon={annotationsIcon}
+      />
     </div>
   );
 };
 
+type FeatureCountProps = {
+  count: number;
+  icon: string;
+  label: string;
+};
+
+const FeatureCount = (props: FeatureCountProps) => (
+  <div className={styles.feature}>
+    <div className={styles.featureImg}>
+      <img src={props.icon} alt="" />
```

Comment:
is there any benefit to having an empty alt over not having an alt at all?

_(especially since here the props.label can be used as an alt)_

Reply:
Agree, `label` is a good one for `alt`.

Reply:
✔️